wazuh/wazuh
# Avoid unrequired changes on ossec.conf when upgrading Windows Agents

|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/5085|

## Description

There is a script called `src/win32/InstallerScripts.vbs` which makes some modifications to the default ossec.conf on certain circumstances. In it, there is a block of code that was intended to enable syscheck bu default in a fresh installation. Nevertheless, the way to check if it isn't being executed on an upgrade was tricky and could produce weird behaviors.

We have removed that code block. Instead of using it, we will just enable syscheck in the default configuration used for both install and upgrade. Also, we will avoid unnecessarily writes to the file. Another block that writes the changes required to ossec.conf (even if there are no required changes at all) have been moved into a conditional structure so te ossec.conf is only written when there are any required changes to do.

## Configuration options

None.

## Logs/Alerts example

![image](https://user-images.githubusercontent.com/15269938/94568043-57eff800-026c-11eb-988c-1026ebf78c6f.png)


## Tests

Jenkins test: packages building, installation, etc...

Manually tested: 
- Fresh install on Windows agent.
- Upgrade from the previous version to 4.0
- Check that syscheck is enabled by default


- Compilation without warnings in every supported platform
- [x] Package installation
- [x] Package upgrade
- [x] QA templates contemplate the added capabilities


Also, I've tested the installation with NSI script.

![image](https://user-images.githubusercontent.com/15269938/94794926-e97e7780-03dc-11eb-8c82-e9fa06153ce7.png)

![image](https://user-images.githubusercontent.com/15269938/94795981-6c540200-03de-11eb-84be-ecab55abec78.png)
